### PR TITLE
Update SignupController@update to include authorization logic.

### PR DESCRIPTION
**Pull request by chloealee**
#### What's this PR do?
- Updates SignupController@update to include authorization logic.
  - Only allows admins and owner of signup to update signup. 
- Updates tests. 

#### How should this be reviewed?
👀 

#### Relevant tickets
Fixes [this](https://www.pivotaltracker.com/n/projects/2019429/stories/153365904) Pivotal Card.

#### Checklist
- [ ] Documentation added for new features/changed endpoints.
- [ ] Tested on staging.
- [ ] Pinged a PM if this is a larger PR that would benefit from some additional testing love.

### CODE REVIEW

File: tests/Http/Three/SignupTest.php [hunk 1]
```diff
@@ -268,19 +269,24 @@ public function testValidationForUpdatingASignup()
     }
 
     /**
-     * Test that non-authenticated user's/apps can't update signups.
+     * TTest that a non-admin or user that doesn't own the signup can't update signup.
```
Comment (sbsmith86) [inline tests/Http/Three/SignupTest.php]: "TTest" 😬

Comment (chloealee) [reply tests/Http/Three/SignupTest.php]: yikes - thanks for catching that! not supposed to be a stutter :)

File: app/Http/Controllers/Three/SignupsController.php [hunk 2]
```diff
@@ -43,7 +44,6 @@ public function __construct(SignupService $signups)
         $this->transformer = new SignupTransformer;
 
         $this->middleware('auth:api', ['only' => ['store', 'update', 'destroy']]);
-        $this->middleware('role:admin', ['only' => ['store', 'update', 'destroy']]); // @TODO: Allow anyone to use this.
```
Comment (DFurnes) [inline app/Http/Controllers/Three/SignupsController.php]: I think we still want `role:admin` on the destroy endpoint!

Comment (chloealee) [reply app/Http/Controllers/Three/SignupsController.php]: @DFurnes added back in in [this PR](https://github.com/DoSomething/rogue/pull/591)!

**Pull request merged**